### ISSUE TITLE
Validate() function in type definition.

### ISSUE THREAD

**Issue by larsvankleef** [bijoutrouvaille/fireward#6]
I'am trying out this compiler to see if it is a replacement for bolt. But i'm bumping into a problem where bolt supports a validation function to defined inside a type definition. I'am using this for users to prevent editing accounts that they dit not create for example:

```
type Account {
  validate() {
    (
      (prior(this) != null && unchanged(this.createdBy)) ||
      (prior(this) == null && this.createdBy == auth.uid)
    )
  }
  
  createdBy: UserId
  displayName: string,
  ...
}
```

is there a work around with this compiler? or is this something that could be added to the compiler?

Source:
[data-validation](https://github.com/FirebaseExtended/bolt/blob/master/docs/guide.md#data-validation)

**Comment by bijoutrouvaille** [comment]
Hi, sorry for the late reply—I've completely missed this. The validation function is not supported yet. When I was using it with Bolt, I ran into so many conceptual difficulties that made the already difficult debugging even more difficult, that I originally gave a very low priority to this functionality.

Lately, I've been leaning back towards it, however. To help me gauge the demand for it, I'll leave this thread open for people to comment with their use cases.

**Comment by meizon** [comment]
I have the need for data validation for: 

- value being larger than another
- value being from a certain list of values

**Comment by bijoutrouvaille** [comment]
Feature is released in version `1.3.1`. Please see the [Complete Example](https://github.com/bijoutrouvaille/fireward#complete-example) in the readme and report any issues that pop up.

@meizon for your second use case, take a look also at literal types.
